Docs : Add External Scripts

physcocode opened the pull request:
Doc is broken down into following 3 parts:
1) Using an external Vue Plugin with Gridsome
2) Using an external library with Gridsome
3) Using external script with Gridsome that does not support Server Side Rendering

tomtev commented:
Thank you :) This is good! Should we tell something about how to import external scripts to components also? That's best practice for proper code-splitting.

tomtev commented:
Btw, we will soon move gridsome.org into the core repo. So your contributions here will be transferred to that.

physcocode commented:
@tomtev I have fixed the docs for rendering non SSR compatible libraries and have also added docs for loading external plugins inside your component as you suggested

tomtev commented:
Could you fix the headlines? 
The first headline should be h1 (one #) and have the "Add external scripts"  title.
The rest should be h2, h3 etc.

This is how it looks now: https://deploy-preview-68--gridsome-org-live.netlify.com/docs/assets-scripts

tomtev commented:
Code indenting in the examples are also a bit wrong. Try use 2x spaces for indenting.

physcocode commented:
let me run beautify on code ones

physcocode commented:
@tomtev merging docs with main repo will be awesome , that would encourage people to contribute to the docs more

thisislawatts commented on docs/assets-scripts.md:
```diff
@@ -1 +1,193 @@
+# Add External Scripts
+It is really easy to use any external javascript with gridsome. Being a Vue based framework any method of importing external scripts in vue works out of the box with Gridsome
+
+## Defining external depedencies Globally
```
The correct spelling here should be `dependencies`

thisislawatts commented on docs/assets-scripts.md:
````diff
@@ -1 +1,193 @@
+}
+```
+
+## Defining external depedencies locally for a component
````
The correct spelling here should be `dependencies`

thisislawatts commented on docs/assets-scripts.md:
````diff
@@ -1 +1,193 @@
+
+Example:
+```javascript
+//MyComponet.vue
````
The correct spelling here should be `Component`

thisislawatts commented on docs/assets-scripts.md:
````diff
@@ -1 +1,193 @@
+
+Example:
+```javascript
+//MyComponet.vue
````
The correct spelling here should be `Component`

thisislawatts commented on docs/assets-scripts.md:
````diff
@@ -1 +1,193 @@
+```
+
+## Using external script with Gridsome that does not support Server Side Rendering
+While `gridsome build` gridsome uses server side rendering to create a fully rendererd page. So if your vue component does not support SSR or your external library like `jquery` changes the dom element it won't be rendered properly. For these type of component we suggest you to bind the component inside `<ClientOnly></ClientOnly>` tag and import library inside vue's `mounted()` function.
````
The correct spelling here is `rendered`

thisislawatts commented on docs/assets-scripts.md:
````diff
@@ -1 +1,193 @@
+</script>
+```
+
+Note: You should avoid using external library like `jquery` that manupulates the DOM and try using a Vue Plugin insted.
````
The correct spellings here are `manipulates` and `instead`

physcocode commented:
@thisislawatts thank you for pointing out I have made the necessary changes :)

tomtev commented on docs/assets-scripts.md:
```diff
@@ -1 +1,193 @@
+  // Set default layout as a global component
+  Vue.component('Layout', DefaultLayout)
+  //Use VueTypedJs plugin inside our project
+  Vue.use(VueTypedJs)
```
It's a bad practice to make a plugin globally available. Maybe we should just show how to add it to a component? Or move the docs about components here up.

physcocode commented:
@tomtev removing the docs for global completely would be a bad idea, as for my own use case the plugin I use is globally defined as it's used in almost all pages with 1 or 2 exception defining again and again would seem a bad practice for me as a part of DRY principal 

Although +1 for moving it up and maybe keeping a note that we should avoid global injection of dependencies and use it only if it's required on more than 2-3 pages

tomtev commented:
@physcocode Yes, that would be perfect :)

physcocode commented:
@tomtev done 💯 💯 💯

tomtev commented:
@physcocode Did some updates to the main titles. I tried to simplify them a bit so they look better in the sidebar. Does it work for you?

physcocode commented:
@tomtev this is perfect, I was thinking same :)

tomtev commented:
Thanks :) Will merge this now

physcocode commented:
Finally 💯 💯

Jasperfr74 commented:
@physcocode, Hi sorry to disturb you  ..  but i'am currently stuck as well with this SSR issue window is not defined during the build .. I cannot figure it out how I can make it work with leaflet plugin ...
I've wrap my component into ClientOnly but itis not enough .. could you help me this issue please ?  Thanks a lot !

7:07:04 PM: ReferenceError: window is not defined
7:07:04 PM:     at assets/js/app.164f23d4.js:4961:17
7:07:04 PM:     at module.exports.version (node_modules/leaflet/dist/leaflet-src.js:7:61)
7:07:04 PM:     at Object.<anonymous> (node_modules/leaflet/dist/leaflet-src.js:10:1)

physcocode commented:
@Jasperfr 74 try this example , in your main.js

if (process.isClient) {
    const ScrollReveal = require('vue-scroll-reveal').default;
    
    Vue.use(ScrollReveal);
}

This example is for ScrollReveal you can use it in any of your dependencies that relow on "window" variable
